File: app/src/main/java/pl/mosenko/sunnypodlaskie/mvp/weatherdatadetail/WeatherDataDetailPresentationModel.java
package pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail;

import pl.mosenko.sunnypodlaskie.util.WeatherDataUtil;

/**
 * Created by syk on 06.06.17.
 */

public class WeatherDataDetailPresentationModel {
    private String titleDetails;
    private String temperature;
    private String description;
    private int getIconResource;
    private String pressure;
    private String windDetails;
    private String humidity;
    private String sunrise;
    private String sunset;

    public String getTitleDetails() {
        return titleDetails;
    }

    public void setTitleDetails(String titleDetails) {
        this.titleDetails = titleDetails;
    }

    public String getTemperature() {
        return temperature;
    }

    public void setTemperature(String temperature) {
        this.temperature = temperature;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getGetIconResource() {
        return getIconResource;
    }

    public void setGetIconResource(int getIconResource) {
        this.getIconResource = getIconResource;
    }

    public String getPressure() {
        return pressure;
    }

    public void setPressure(String pressure) {
        this.pressure = pressure;
    }

    public String getWindDetails() {
        return windDetails;
    }

    public void setWindDetails(String windDetails) {
        this.windDetails = windDetails;
    }

    public String getHumidity() {
        return humidity;
    }

    public void setHumidity(String humidity) {
        this.humidity = humidity;
    }

    public String getSunrise() {
        return sunrise;
    }

    public void setSunrise(String sunrise) {
        this.sunrise = sunrise;
    }

    public String getSunset() {
        return sunset;
    }

    public void setSunset(String sunset) {
        this.sunset = sunset;
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/mvp/weatherdatadetail/WeatherDataDetailPresentationModel.kt
package pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail

import pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity
import pl.mosenko.sunnypodlaskie.util.WeatherDataUtil
import pl.mosenko.sunnypodlaskie.util.WeatherDataUtil.DEFAULT_PLACEHOLDER

/**
 * Created by syk on 06.06.17.
 */
class WeatherDataDetailPresentationModel
   private constructor (
    val titleDetails: String,
    val temperature: String,
    val description: String? = null,
    val iconResource: Int,
    val pressure: String? = null,
    val windDetails: String? = null,
    val humidity: String? = null,
    val sunrise: String? = null,
    val sunset: String? = null
) {
    constructor(weatherDataEntity: WeatherDataEntity) : this(
        titleDetails = WeatherDataUtil.getDetailsTitle(
            weatherDataEntity.city,
            weatherDataEntity.receivingTime
        ),
        temperature = WeatherDataUtil.getFormattedTemperature(weatherDataEntity.temperature),
        description = weatherDataEntity.weatherCondition?.description ?: DEFAULT_PLACEHOLDER,
        iconResource = WeatherDataUtil.getWeatherIconResourceByCode(weatherDataEntity.iconKey),
        pressure = WeatherDataUtil.getFormattedPressure(weatherDataEntity.pressure),
        windDetails = WeatherDataUtil.getFormattedWindDetails(
            weatherDataEntity.windSpeed,
            weatherDataEntity.windDegree
        ),
        humidity = WeatherDataUtil.getFormattedHumidity(weatherDataEntity.humidity),
        sunrise = WeatherDataUtil.getFormattedTime(weatherDataEntity.sunrise),
        sunset = WeatherDataUtil.getFormattedTime(weatherDataEntity.sunset)
    )
}
